/**
 Copyright 2022-2023 PrimeQA Team

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
*/

syntax = "proto3";
import "parameter.proto";
import "retriever.proto";

package rerank;

/**
    Service to rerank documents given a query
*/
service RerankerService {
    rpc GetRerankers(GetRerankersRequest) returns (GetRerankersResponse);
    rpc Rerank(RerankRequest) returns (RerankResponse);
}

message Reranker {
    string reranker_id = 1;
    repeated parameter.Parameter parameters = 2;
}

message GetRerankersRequest {
}

message GetRerankersResponse {
    repeated Reranker rerankers = 1;
}

message RerankRequest {
    Reranker reranker = 1;
    repeated string queries = 2;
    repeated retrieve.HitPerQuery hitsperquery = 3;
    optional string index_id = 4; 
}

message RerankResponse {
    repeated retrieve.HitPerQuery hits = 1;
}